yjit/src/core.rs

```diff
@@ -1234,10 +1234,20 @@ fn make_branch_entry(block: BlockRef, src_ctx: &Context, gen_fn: BranchGenFn) ->
     return branchref;
 }
 
-// Called by the generated code when a branch stub is executed
-// Triggers compilation of branches and code patching
-#[no_mangle]
-pub extern "C" fn rb_yjit_rust_branch_stub_hit(branch_ptr: *const c_void, target_idx: u32, ec: EcPtr) -> *const u8
+/// Generated code calls this function with the SysV calling convention.
+/// See [get_branch_target].
+extern "sysv64" fn branch_stub_hit(
+    branch_ptr: *const c_void,
+    target_idx: u32,
+    ec: EcPtr
+) -> *const u8
+{
+    with_vm_lock(src_loc!(), || branch_stub_hit_body(branch_ptr, target_idx, ec))
```

Comment:
I considered making the whole thing a macro but leaned towards this being a plain function call because code wrapped inside macros are harder for rust-analyzer and rustfmt to understand and then they tend to give up. Macros have freedom to do too much weird things I guess.

I anticipate we have like only a handful of usages of this function so I would prefer this uglier but easier-to-analyze setup.